libmount/src/utils.c

```diff
@@ -266,6 +266,63 @@ int mnt_is_readonly(const char *path)
 	return 0;
 }
 
+static int get_mnt_id(	int fd, const char *path,
+			uint64_t *uniq_id, int *id)
+{
+#if defined(HAVE_STATX) && defined(HAVE_STRUCT_STATX) && defined(HAVE_STRUCT_STATX_STX_MNT_ID)
+	int rc;
+	struct statx sx = { 0 };
+	int flags;
+
+	flags = AT_STATX_DONT_SYNC | AT_NO_AUTOMOUNT;
+	if (!path || !*path)
+		flags |= AT_EMPTY_PATH;
+
+	if (id) {
+		rc = statx(fd,	path && *path ? path : "", flags,
+				STATX_MNT_ID, &sx);
+		if (rc)
+			return rc;
+		*id = sx.stx_mnt_id;
+	}
+	if (uniq_id) {
+# ifdef STATX_MNT_ID_UNIQUE
```

Comment:
Why not add a fallback define for `STATX_MNT_ID_UNIQUE`?
`statx()` would return `EINVAL` in that case.